kube-timer-instruction.md

```diff
@@ -0,0 +1,153 @@
+| -a | Create all the resources together and wait for the previous resouces to be ready. The time of single resource creation would contain waiting in queue time. Without this parameter, the resources are created one by one; the next resource will only be created once the previous one is ready.|
+| -d | Delete the resource |
+| -n | Name of the resource to be leleted |
+| -D | Delete all the resources together | 
+| --started-event-reason | The reason of event. It can be used as the start time of provision/deletion |
+| --finished-event-reason | The reason of event. It can be used as the end time of provision/deletion |
+| --namespace | The namespace of resources | 
+
+## Samples for the kube-timer
+
+### Get the provision time of single LoadBalancer service.
+1. Create a file named `svc.yaml` 
+
+2. Sample YAML definition:
+
+```yml
+kind: Service
+apiVersion: v1
+metadata:
+  Name: testsvc
+  namespace: default
+spec:
+  selector:
+    app: nsg-test
+  ports:
+    - name: port1
+      port: 81
+      targetPort: 81
+      protocol: TCP
+  type: LoadBalancer
+  
+```
+
+3. Get the provision time of this LoadBalancer service.
+```shell
+kube-timer svc -f <service definition yaml file>
+```
+
+### Get the provision time of multiple LoadBalancer services.
+
+#### Do not contain waiting-in-queue time
+
+As we do not specify `-a`, all the services will be created one by one. The next service is only be created once the previous one is ready.
+1. Create a file named `svc.yaml` 
+
+2. Sample YAML definition:
+
+```yml
+kind: Service
+apiVersion: v1
+metadata:
+  generateName: lb-svc-internal- # All service names are prefixed with lb-svc-internal-
+  namespace: default
+  # annotations:
+  #   service.beta.kubernetes.io/azure-pip-tags: "a=b,c=d"
+  # service.beta.kubernetes.io/azure-load-balancer-internal: "true"
+  #   service.beta.kubernetes.io/azure-load-balancer-internal-subnet: "lb-subnet"
+  #   service.beta.kubernetes.io/azure-load-balancer-mixed-protocols: "true"
+  #   service.beta.kubernetes.io/azure-load-balancer-mode: "test-capz-2-vmss-0"
+  #   service.beta.kubernetes.io/azure-shared-securityrule: "true"
+  #   service.beta.kubernetes.io/azure-deny-all-except-load-balancer-source-ranges: "true"
+  #   service.beta.kubernetes.io/azure-dns-label-name: "test-dns-lb-svc-1-managed"
+spec:
+  selector:
+    app: nsg-test
+  ports:
+    - name: port1
+      port: 81
+      targetPort: 81
+      protocol: TCP
+    # - name: port2
+    #   port: 82
+    #   targetPort: 82
+    #   protocol: UDP
+  type: LoadBalancer
+  # loadBalancerIP: 20.252.52.147
+  # ipFamilies:
+  # - IPv4
+  # - IPv6
+```
+
+3. Get the provision time of each service and total time for the whole provision.
+
+```shell
+kube-timer svc -f <service definition yaml file> -c <number of services>
+```
+
+4.Sample output:
+![image](https://user-images.githubusercontent.com/45681473/177050104-b3a7fa1e-08ad-4811-a1b2-fc6c83e15101.png)
+
+
+#### Contains waiting-in-queue time
```

Comment:
Again, we need to put it clear that this behavior varies from each cloud provider.